compair/api/__init__.py

```diff
@@ -208,6 +212,27 @@ def file_retrieve(file_type, file_name):
             'report': app.config['REPORT_FOLDER']
         }
         file_path = '{}/{}'.format(file_dirs[file_type], file_name)
+        params = attachment_download_parser.parse_args()
+
+        if file_type == 'attachment':
+            attachment = File.get_by_file_name_or_404(
+                file_name,
+                joinedloads=['answers']
+            )
+
+            for answer in attachment.answers:
+                require(READ, answer,
+                    title="Attachment Unavailable",
+                    message="Sorry, your role does not allow you to view the attachment.")
+
```

Comment:
just thought that we can add `for assignment in attachment.assignments:` bouncing here too (for assignment attachments)